src/odemis/util/test/img_test.py

```diff
@@ -837,6 +837,105 @@ def test_25d(self):
         self.assertEqual(255, out[128, 256, 3])
 
 
+class TestImageIntegrator(unittest.TestCase):
+
+    def setUp(self):
+        dtype = numpy.dtype("uint16")
+        im = numpy.ones((5, 5), dtype=dtype)
+        metadata = {model.MD_ACQ_DATE: time.time(),
+                    model.MD_BPP: 12,
+                    model.MD_BINNING: (1, 1),  # px, px
+                    model.MD_PIXEL_SIZE: (1e-6, 2e-5),  # m/px
+                    model.MD_POS: (1e-3, -30e-3),  # m
+                    model.MD_EXP_TIME: 1.2,  # s
+                    model.MD_DET_TYPE: model.MD_DT_INTEGRATING,
+                    model.MD_DWELL_TIME: 1e-06,  # s
+                    }
+        self.data = model.DataArray(im, metadata)
+        self.integrated_data = None
+        self.img_intor = None
+        self.integrationCounts = 3
+
+    def test_simple(self):
+        self.img_intor = img.ImageIntegrator(self.integrationCounts)
+        self.assertEqual(self.img_intor.steps, self.integrationCounts)
+
+        for i in range(self.integrationCounts):
+            self.integrated_data = self.img_intor.append(self.data[i])
```

Comment:
I'd recommend to test passing data of 2D. (data[i] is only one dimension). As data contains only 1's, I'd suggest to just use `data`, and it's already 2D.